libdnf/base/goal.cpp

```diff
@@ -0,0 +1,551 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/base/goal.hpp"
+
+#include "../rpm/package_set_impl.hpp"
+#include "../rpm/solv/goal_private.hpp"
+#include "../rpm/solv/id_queue.hpp"
+#include "../rpm/solv/package_private.hpp"
+#include "../rpm/solv_query_impl.hpp"
+#include "../rpm/solv_sack_impl.hpp"
+#include "../utils/utils_internal.hpp"
+
+#include "libdnf/rpm/solv_query.hpp"
+
+
+namespace {
+
+static void add_obseletes(const libdnf::rpm::SolvQuery & base_query, libdnf::rpm::PackageSet & data) {
```

Comment:
Users sometimes discuss what to use? `static` or unnamed/anonymous `namespace`?
You found a solution. You are using the keyword `static` in an anonymous namespace. :-)